app/profile/page.tsx

```diff
@@ -0,0 +1,299 @@
+    enabled: !!address,
+  });
+
+  const filterActiveBets = useMemo(
+    () =>
+      userPositions?.filter(
+        userPosition => !userPosition.position.conditions[0].resolved
+      ) ?? [],
+    [userPositions]
+  );
+
+  const filterCompleteBets = useMemo(
+    () =>
+      userPositions?.filter(
+        userPosition => userPosition.position.conditions[0].resolved
+      ) ?? [],
+    [userPositions]
+  );
+
+  const { data: aiAgentsList } = useQuery({
+    queryKey: ['getAIAgents'],
+    queryFn: getAIAgents,
+    staleTime: Infinity,
+  });
+
+  const isAIAgent =
+    aiAgentsList?.some(aiAgent => String(aiAgent.address).toLowerCase() === address) ??
+    false;
+
+  if (!address || !isAddress(address)) return null;
+
+  const isLoadingUserPositions = userPositions === undefined;
+
+  const spentAmountInUSD = userPositions?.reduce((acc, userPosition) => {
+    const market = new Market(userPosition.fpmm);
+
+    if (market.answer === null) return acc;
+
+    const amountSpentInUSD = tradesCollateralAmountUSDSpent({
+      fpmmTrades: userPosition.fpmmTrades,
+    });
+
+    return amountSpentInUSD + acc;
+  }, 0);
+
+  const winAmountInUSD = userPositions?.reduce((acc, userPosition) => {
+    const position = new Position(userPosition.position);
+    const market = new Market(userPosition.fpmm);
+
+    if (market.answer === null || market.isLoser(position.outcomeIndex - 1)) return acc;
+
+    // missing usd price
+    const winOutcomeBalancesInColleteralToken = tradesOutcomeBalance({
+      fpmmTrades: userPosition.fpmmTrades,
+    });
+
+    return winOutcomeBalancesInColleteralToken + acc;
+  }, 0);
+
+  const potentialWinAmountInUSD = userPositions?.reduce((acc, userPosition) => {
+    const position = new Position(userPosition.position);
+    const market = new Market(userPosition.fpmm);
+
+    if (!userPosition.balance || market.answer !== null || !market.hasLiquidity)
+      return acc;
+
+    const oneShareSellPrice = calcSellAmountInCollateral(
+      userPosition.balance,
+      market.fpmm.outcomeTokenAmounts,
+      position.outcomeIndex - 1,
+      parseFloat(formatEther(market.fpmm.fee))
+    );
+
+    if (!oneShareSellPrice) return acc;
+
+    // missing usd price
+    const priceInColleteralToken = parseFloat(formatEther(oneShareSellPrice));
+
+    return priceInColleteralToken + acc;
+  }, 0);
+
+  const totalVolumeInUSD = userPositions?.reduce(
+    (acc, userPosition) =>
+      tradesVolume({
+        fpmmTrades: userPosition.fpmmTrades,
+      }) + acc,
+    0
+  );
+
+  const numberOfWonBets = userPositions?.reduce((acc, userPosition) => {
+    const position = new Position(userPosition.position);
+    const market = new Market(userPosition.fpmm);
+
+    if (market.answer !== null && market.isWinner(position.outcomeIndex - 1))
+      return acc + 1;
+
+    return acc;
+  }, 0);
+
+  const numberOfClosedBets = userPositions?.reduce((acc, userPosition) => {
```

Comment:
same, wondering if would be better to have this logic on market instead on component

Reply:
https://github.com/SwaprHQ/presagio/pull/83#discussion_r1745598437 answered here